app/src/main/java/org/simple/clinic/teleconsultlog/drugduration/DrugDurationSheet.kt

```diff
@@ -0,0 +1,165 @@
+package org.simple.clinic.teleconsultlog.drugduration
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import com.jakewharton.rxbinding3.widget.editorActions
+import io.github.inflationx.viewpump.ViewPumpContextWrapper
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.ofType
+import kotlinx.android.synthetic.main.sheet_drug_duration.*
+import org.simple.clinic.ClinicApp
+import org.simple.clinic.R
+import org.simple.clinic.ReportAnalyticsEvents
+import org.simple.clinic.di.InjectorProviderContextWrapper
+import org.simple.clinic.drugs.PrescribedDrug
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.teleconsultlog.drugduration.di.DrugDurationComponent
+import org.simple.clinic.util.LocaleOverrideContextWrapper
+import org.simple.clinic.util.unsafeLazy
+import org.simple.clinic.util.wrap
+import org.simple.clinic.widgets.BottomSheetActivity
+import org.simple.clinic.widgets.setTextAndCursor
+import org.simple.clinic.widgets.textChanges
+import java.util.Locale
+import javax.inject.Inject
+
+class DrugDurationSheet : BottomSheetActivity(), DrugDurationUi, DrugDurationUiActions {
+
+  @Inject
+  lateinit var locale: Locale
+
+  @Inject
+  lateinit var effectHandlerFactory: DrugDurationEffectHandler.Factory
+
+  companion object {
+    private const val EXTRA_DRUG_UUID = "prescribedDrugUuid"
+    private const val EXTRA_DRUG_NAME = "prescribedDrugTitle"
+    private const val EXTRA_DRUG_DOSAGE = "prescribedDrugDosage"
+    private const val EXTRA_DURATION = "drugDuration"
+    private const val EXTRA_SAVED_DRUG_UUID = "savedDrugUuid"
+    private const val EXTRA_SAVED_DURATION = "savedDrugDuration"
+
+    fun intent(
+        context: Context,
+        drug: PrescribedDrug,
+        drugDuration: String
+    ): Intent {
+      return Intent(context, DrugDurationSheet::class.java).apply {
+        putExtra(EXTRA_DRUG_UUID, drug.uuid)
+        putExtra(EXTRA_DRUG_NAME, drug.name)
+        putExtra(EXTRA_DRUG_DOSAGE, drug.dosage)
+        putExtra(EXTRA_DURATION, drugDuration)
```

Comment:
We don't need to do this in this PR, but I can suggest that we extract the arguments required for this screen into their own class and save that as a single `Parcelable` argument.